dialogue-core/src/main/java/com/palantir/dialogue/core/QueuedChannel.java

```diff
@@ -109,6 +84,16 @@
      */
     @Override
     public ListenableFuture<Response> execute(Endpoint endpoint, Request request) {
+        // Optimistically avoid the queue in the fast path.
+        // Queuing adds contention between threads and should be avoided unless we need to shed load.
```

Comment:
you now can queue jump in a pretty weird way? probably need to at least check the queue is empty

Reply:
An empty check would also result in a race. I don't think we make any guarantees that the first request submitted will be the first one we attempt to execute.

Reply:
I guess the situation that is now possible (albeit unlikely) is that I submit a request that gets queued and every subsequent request preempts it such that it is never run. That seems like an undesirable structural thing?